src/App.tsx

```diff
@@ -16,47 +22,84 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [activeButton, setActiveButton] = useState<SortType | null>(null);
+  const [resetVisible, setResetVisible] = useState(false);
```

Comment:
you'd better make those states
```suggestion
  const [sortType, setSortType] = useState<SortType | null>(null);
  const [isReversed, setIsReversed] = useState(false);
```